File: String_Matching_in an_Array.py

words = ["mass","as","hero","superhero"]
def stringMatching(words):
    ls = []
    for i in range(len(words)):
        for j in range(len(words)):
            if words[i] == words[j]:
                continue
            if words[i] in words[j]:
                ls.append(words[i])
    return ls
print(stringMatching(words))

File: String_Matching_in an_Array/String_Matching_in an_Array.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	words := []string{"mass", "as", "hero", "superhero"}
	fmt.Println(stringMatching(words))
}

func stringMatching(words []string) []string {
	var res []string
	for _, i := range words {
		for _, j := range words {
			if i == j {
				continue
			}
			if strings.Contains(j, i) {
				res = append(res, i)
				break
			}
		}
	}
	return res
}
